### PR TITLE
Added settings description files

### PR DESCRIPTION
**Pull request by ianf77**
Cherry-pick into:

* [X] Foreman 3.1
* For Foreman 3.0 or older, file a separate PR request

### CODE REVIEW

**Comment by ianf77** [comment]
@lmcgarry Can you peer review?

File: guides/common/modules/ref_facts-settings.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[id="facts_settings_{context}"]
+= Facts Settings
+
+[cols="30%,30%,40%",options="header"]
+|====
+| Setting | Default Value | Description
+| *Create new host when facts are uploaded* | Yes | {Project} creates the host when new facts are received.
+| *Location fact* | {project_context}_location | Hosts created after a Puppet run are placed in the location specified by this fact.
```
Comment (Lennonka) [inline guides/common/modules/ref_facts-settings.adoc:8]: It should be `{project-context}` instead of `{project_context}`, all occurences

**Review by Lennonka — CHANGES_REQUESTED** [review]
Syntactic review :) I didn't do any fact checking nor phrasing review.

File: guides/common/modules/ref_template-sync-settings.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+| *Lock templates* | Keep, do not lock new | How to handle lock for imported templates.
+| *Metadata export mode* | Refresh | Default metadata export mode.
+
+Possible options:
+
+* `refresh` re-renders metadata.
+
+* `keep` keeps existing metadata.
+
+* `remove` exports the template without metadata.
```
[the remark this answers is not in the document]
Comment (ianf77) [reply guides/common/modules/ref_template-sync-settings.adoc:24]: Actually no. I'd forgotten that bullet points don't work very well in tables.

File: guides/common/modules/ref_template-sync-settings.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+| *Repo* | | Target path from where to import or export templates.
+Different protocols can be used, for example:
+
+* `/tmp/dir`
+
+* `git://example.com`
+
+* \https://example.com
+
+* `ssh://example.com`
```
[the remark this answers is not in the document]
Comment (ianf77) [reply guides/common/modules/ref_template-sync-settings.adoc:36]: Removed bullet point markers. They don't always work in tables.

**Comment by Lennonka** [comment]
@ianf77 I checked and some comments haven't been addressed yet.

File: guides/common/modules/ref_authentication-settings.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+| *OAuth active* | Yes | {Project} will use OAuth for API authorization.
+| *OAuth consumer key* | \\***** | OAuth consumer key.
+| *OAuth consumer secret* | \\***** | OAuth consumer secret.
+| *OAuth map users* | No | {Project} maps users by username in the request-header.
+If this is disabled, OAuth requests have administrator rights.
+| *Failed login attempts limit* | 30 | {Project} blocks user logins from an incoming IP address for 5 minutes after the specified number of failed login attempts.
+Set to 0 to disable brute force protection.
+| *Restrict registered {SmartProxies}* | Yes | Only known {SmartProxies} can access features that use {SmartProxy} authentication.
+ifdef::satellite[]
+| *Require SSL for capsules* | Yes | Client SSL certificates are used to identify {SmartProxies} (:require_ssl should also be enabled).
```
Comment (Lennonka) [inline guides/common/modules/ref_authentication-settings.adoc:16]: Code format `:require_ssl`

File: guides/common/modules/ref_authentication-settings.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+| *SSL CA file* | `/etc/foreman/proxy_ca.pem` | SSL CA file path that {Project} uses to communicate with its proxies.
+| *SSL private key* | `/etc/foreman/client_key.pem` | SSL Private Key path that {Project} uses to communicate with its proxies.
+| *SSL client DN env* | HTTP_SSL_CLIENT_S_DN | Environment variable containing the subject DN from a client SSL certificate.
+| *SSL client verify env* | HTTP_SSL_CLIENT_VERIFY | Environment variable containing the verification status of a client SSL certificate.
+| *SSL client cert env* | HTTP_SSL_CLIENT_CERT | Environment variable containing a client's SSL certificate.
+| *Server CA file* | | SSL CA file path used in templates to verify the connection to {Project}.
+| *Websockets SSL key* | `etc/pki/katello/private/katello-apache.key` | Private key file path that {Project} uses to encrypt websockets.
+| *Websockets SSL certificate* | `/etc/pki/katello/certs/katello-apache.crt` | Certificate path that {Project} uses to encrypt websockets.
+| *Websockets encryption* | Yes | VNC/SPICE websocket proxy console access encryption (websockets_ssl_key/cert setting required).
```
Comment (Lennonka) [inline guides/common/modules/ref_authentication-settings.adoc:28]: Code format `websockets_ssl_key/cert` somehow?

File: guides/common/modules/ref_authentication-settings.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+| *Login delegation logout URL* | | Redirect your users to this URL on logout.
+Enable *Authorize login delegation* also.
+| *Authorize login delegation auth source user autocreate* | External | Name of the external authentication source where unknown externally authenticated users (see *Authorize login delegation*) are created.
+Empty means no autocreation.
+| *Authorize login delegation* | No | Authorize login delegation with REMOTE_USER HTTP header.
```
Comment (Lennonka) [inline guides/common/modules/ref_authentication-settings.adoc:33]: Code format header name

File: guides/common/modules/ref_authentication-settings.adoc [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+| *Authorize login delegation API* | No | Authorize login delegation with REMOTE_USER HTTP header for API calls too.
```
Comment (Lennonka) [inline guides/common/modules/ref_authentication-settings.adoc:34]: Code format header name

File: guides/common/modules/ref_authentication-settings.adoc [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+| *Idle timeout* | 60 | Log out idle users after the specified number of minutes.
+| *BCrypt password cost* | 9 | Cost value of bcrypt password hash function for internal auth-sources (4-30).
+A higher value is safer but verification is slower, particularly for stateless API calls and UI logins.
+A password change is needed to affect existing passwords.
+| *BMC credentials access* | Yes | Permits access to BMC interface passwords through ENC YAML output and in templates.
+| *OIDC JWKs URL* | | OpenID Connect _JSON Web Key Set_ (JWKS) URL.
+Typically \https://keycloak.example.com/auth/realms/<realm name>/protocol/openid-connect/certs when using Keycloak as an OpenID provider.
```
Comment (Lennonka) [inline guides/common/modules/ref_authentication-settings.adoc:41]: Code format URL? Will the text be line-broken if we do this or will it force the table to be wider?

File: guides/common/modules/ref_provisioning-settings.adoc [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+The `Off` option disables the name generator function and leaves the hostname field blank.
+| *Default PXE global template entry* | | Default PXE menu item in a global template - `local`, `discovery` or custom, use blank for template default.
+| *Default PXE local template entry* | | Default PXE menu item in local template - `local`, `local_chain_hd0`, or custom, use blank for template default.
+| *iPXE intermediate script* | iPXE intermediate script | Intermediate iPXE script for unattended installations.
+| *Destroy associated VM on host delete* | No | Destroy associated VM on host delete.
+When enabled, VMs linked to hosts are deleted on Compute Resource.
+When disabled, VMs are unlinked when the host is deleted, meaning they remain on Compute Resource and can be re-associated or imported back to {Project} again.
+This does not automatically power off the VM
+| *Maximum structured facts* | 100 | Maximum number of keys in structured subtree, statistics stored in {project-context}::dropped_subtree_facts.
```
Comment (Lennonka) [inline guides/common/modules/ref_provisioning-settings.adoc:44]: Will the attribute be replaced if we code format `{project-context}::dropped_subtree_facts` ?

**Review by Lennonka — CHANGES_REQUESTED** [review]
A couple of formatting suggestions, otherwise looks good!

**Review by Lennonka — APPROVED** [review]
Looking good!

**Pull request merged**